### PR TITLE
Go mod

### PR DESCRIPTION
**Pull request by vaikas**
Changes to use go mod.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+replace (
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.2
+	k8s.io/api => k8s.io/api v0.16.4
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.16.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.16.4
+	k8s.io/client-go => k8s.io/client-go v0.16.4
+	k8s.io/code-generator => k8s.io/code-generator v0.16.4
+	knative.dev/pkg => github.com/chizhg/pkg v0.0.0-20200420011907-9117cd5cf224
```
Comment (mattmoor) [inline go.mod:28]: I don't think we should be checking this in.

File: third_party/VENDOR-LICENSE [hunk 2]
```diff
@@ -1,7324 +0,0 @@
-
```
Comment (chizhg) [inline third_party/VENDOR-LICENSE:1]: This file is now empty. Please follow the below steps to fix it:
1. Run `go get -u knative.dev/test-infra` to update test-infra
2. Add the lines in `/hack/update-deps.sh` like https://github.com/knative/net-http01/pull/19/files#diff-58d4b0ee04af7710eefa27cf91255798R26-R28
3. Run `./hack-update-deps.sh` to regenerate the license file.

File: go.mod (hunk 1 above)
Comment (n3wscott) [reply go.mod:28]: LGTM this and use pkg master: https://github.com/knative/pkg/pull/1251

File: third_party/VENDOR-LICENSE (hunk 2 above)
Comment (vaikas) [reply third_party/VENDOR-LICENSE:1]: Great, thanks! Done.

**Comment by chizhg** [comment]
@googlebot rescan

**Comment by chizhg** [comment]
To fix the build-tests, please add `export GO111MODULE=on` at the top of the `/test/presubmit-tests.sh` file. (These Go modules settings will be moved to `test-infra/scripts/` after the migration is done)

**Comment by chizhg** [comment]
@googlebot rescan

**Comment by chizhg** [comment]
@googlebot rescan

File: go.mod (hunk 1 above)
Comment (chizhg) [reply go.mod:28]: /hold
For this comment.

Comment (vaikas) [reply go.mod]: done ✅

**Pull request merged**